### DISCUSSION TITLE
Whats the difference between Download and CreateSignedURL?

### DISCUSSION THREAD

**Discussion by IHIutch** [supabase/supabase#1232] [Questions] [answered]
Seems like they return similar info, when/why should I use one or the other?

**Comment by gomflo** [comment]
### from.createSignedUrl()

Create signed url to download file without requiring permissions. This URL can be valid for a set number of seconds.

https://supabase.io/docs/reference/javascript/storage-from-createsignedurl

i think thats the main diference.

**Reply by IHIutch** [reply]
I think I understand that from a high level, but I was hoping for more of a layman understanding of when and how I might use each.

**Comment by inian** [comment, marked as answer]
Hi @IHIutch, 

Both serve the same function of viewing the file. The main differences are
- download returns the file directly whereas with signedURL you get the URL and need to make another request to view the file. 
- with a signedURL, you don't need to send any Authorization headers. This makes it easy to just put the URL in an image tag for example. With download you can't do this, since it requires a Authorization header. 
- With signedURL, you don't need to expose even your `anonKey`. If you quickly want to get a URL to your object, you can create a signed URL with a expiry and pass it around.

**Reply by IHIutch** [reply]
This is very helpful, thank you!

**Comment by coderinblack08** [comment]
@inian Quick question, what was the design choice behind having an expiry and token?
